Fix defaults for the traceSampleRate and traceContextInjection in RUM with new mobile SDK releases (#32686)
DataDog/documentation: content/en/real_user_monitoring/guide/mobile-sdk-upgrade.md

{{< tabs >}}
{{% tab "Android" %}}

The transition from version 2 to version 3 focuses on removing support for the legacy Open Tracing project, altering default behaviour for distributed tracing and improving the stability and consistency of the SDK.

{{% /tab %}}

{{% tab "iOS" %}}

> The migration from v2 to v3 focuses on streamlining modules, refining default behaviour, and improving reliability across product features.

Reviewer: ```suggestion
The migration from v2 to v3 focuses on streamlining modules, refining default behavior, and improving reliability across product features.
```

)
```

> We changed the default behaviour of distributed traces. The SDK will now by default inject trace context into outgoing request only when the trace is sampled in.

Reviewer: ```suggestion
We changed the default behavior of distributed traces. The SDK will now by default inject trace context into outgoing requests only when the trace is sampled in.
```

> Refer to [Connect RUM and Traces][5] for more information.

Reviewer: you'll have to add this to line 343 for the link to work:

`[2]:  /real_user_monitoring/correlate_with_other_telemetry/apm`

- `Monitor.removeViewAttribute(forKey:)`
- `Monitor.removeViewAttributes(forKeys:)`

We changed the default behaviour of distributed traces. The SDK will now by default inject trace context into outgoing request only when the trace is sampled in.

> Refer to [Connect RUM and Traces][5] for more information.

Reviewer: similarly, you'll have to add the following to line 450 for the link to work:

`[2]:  /real_user_monitoring/correlate_with_other_telemetry/apm`

|`DDNSURLSessionDelegate()`|`URLSessionInstrumentation.enable(with:)`|

[1]: /real_user_monitoring/session_replay/mobile/privacy_options?platform=ios

> [2]:  /real_user_monitoring/correlate_with_other_telemetry/apm

Reviewer 2: ```suggestion
[2]: /real_user_monitoring/correlate_with_other_telemetry/apm
```
